marlin/src/marlin/marlin.rs

```diff
@@ -512,54 +423,36 @@ impl<
         let evaluations = evaluations_unsorted.iter().map(|x| x.1).collect::<Vec<TargetField>>();
         end_timer!(eval_time);
 
-        if terminator.load(Ordering::Relaxed) {
-            return Err(MarlinError::Terminated);
-        }
+        Self::terminate(terminator)?;
 
-        if MM::RECURSION {
-            fs_rng.absorb_nonnative_field_elements(&evaluations, OptimizationType::Weight);
-        } else {
-            fs_rng.absorb_bytes(&to_bytes_le![&evaluations].unwrap());
-        }
+        fs_rng.absorb_nonnative_field_elements(&evaluations, OptimizationType::Weight);
 
-        let pc_proof = if MM::RECURSION {
-            let num_open_challenges: usize = 5;
+        let num_open_challenges: usize = 7;
```

Comment:
We use individual opening challenges by default, since that's more efficient for our verifiers.